virttest/remote.py

```diff
@@ -1359,16 +1359,28 @@ def run(self, command, timeout=60, ignore_status=False):
 class VMManager(object):
     """Manage VM on remote host"""
 
+    CMD_TIMEOUT = 240
+
     def __init__(self, params):
         self.remote_host = params.get("server_ip")
         self.remote_user = params.get("server_user")
         self.remote_pwd = params.get("server_pwd")
+        self.vm_ip = params.get("vm_ip")
+        self.vm_pwd = params.get("vm_pwd")
+        self.vm_user = params.get("vm_user", 'root')
+        self.port = params.get("port", 22)
+        if not any([self.remote_host, self.remote_user, self.remote_pwd,
+                   self.vm_ip, self.vm_pwd]):
+            raise exceptions.TestError("At least one of [remote_host|"
+                                       "remote_user|remote_pwd|vm_ip|"
+                                       "vm_pwd] is invalid!")
         self.runner = RemoteRunner(host=self.remote_host,
                                    username=self.remote_user,
                                    password=self.remote_pwd)
+        self.cmd_output = self.cmd_output_safe
+        self.cmd = self.cmd_output_safe
 
-    def setup_ssh_auth(self, vm_ip, vm_pwd, vm_user="root",
-                       port=22, timeout=20):
+    def setup_ssh_auth(self):
```

Comment:
I assume this will impact current use cases ,which pass vm_user and vm_pwd directly.

Reply:
@chunfuwen yes. I have updated tp-libvirt cases to adopt the new API. See https://github.com/autotest/tp-libvirt/pull/3021